CMakeLists.txt

```diff
@@ -37,12 +37,17 @@ set(SHARED_UTIL_SRC_FOLDER ${CMAKE_CURRENT_LIST_DIR}/src CACHE INTERNAL "this is
 set(SHARED_UTIL_ADAPTER_FOLDER "${CMAKE_CURRENT_LIST_DIR}/adapters" CACHE INTERNAL "this is where the adapters live" FORCE)
 
 include_directories(${SHARED_UTIL_INC_FOLDER})
-
+get_property(dirs DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
+foreach(dir ${dirs})
+  message(STATUS "dir='${dir}'")
+endforeach()
 CHECK_INCLUDE_FILES(stdint.h HAVE_STDINT_H)
 CHECK_INCLUDE_FILES(stdbool.h HAVE_STDBOOL_H)
 
-if ((NOT HAVE_STDINT_H) OR (NOT HAVE_STDBOOL_H))
+if (WINDOWS)
```

Comment:
There is a problem how cmake checks header files:
It first tries to compile with the `-c` option of gcc. If this succeeds, it tries to link.
And this second step failed, because cmake did not use the linker flags as expected.
(autoconf does it better: It first checks, that the compiler and provided flags works).

Maybe there is something wrong with the CMakeList.txt files of the super project, which uses this project as a sub-module: The linker flags are only set into the Makefiles, if cmake is called twice.

I will modify this pull request, by reducing to commit e661836.